pkg/admission/init.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	"reflect"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/api"
+)
+
+type PluginInitializer interface {
+	Initialize(plugins []Interface)
```

Comment:
One rule them all, hmm? :) I don't want to imagine the initializer once it has 10 arguments you have to pass. As long as the number of arguments is sane and it is obvious what you need to pass, single initializer is good as any other.